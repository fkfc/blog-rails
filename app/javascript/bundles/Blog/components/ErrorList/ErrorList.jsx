import React from 'react';
import PropTypes from 'prop-types';

export default function ErrorList(props) {
  const { errors, name } = props;
  const count = errors.length;
  if (count === 0) return null;

  const title = `${count} ${(count > 1) ? 'errors' : 'error'} prevented this ${name} from saving:`;

  return (
    <div className="alert alert-danger">
      <h5>
        {title}
      </h5>
      <ul>
        {
          errors.map((error, index) => {
            const key = `error${index}`;
            return (<li key={key}>{error}</li>);
          })
        }
      </ul>
    </div>
  );
}

ErrorList.propTypes = {
  errors: PropTypes.arrayOf(PropTypes.string),
  name: PropTypes.string.isRequired,
};

ErrorList.defaultProps = {
  errors: [],
};
