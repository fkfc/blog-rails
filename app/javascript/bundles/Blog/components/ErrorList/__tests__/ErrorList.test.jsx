import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import ErrorList from '../ErrorList';

describe('Testing the Post Component', () => {
  configure({ adapter: new Adapter() });
  let props;
  beforeEach(() => {
    props = {
      name: 'test',
      errors: ['error1', 'error2', 'error3', 'error4'],
    };
  });
  it('Should render correctly', () => {
    const wrapper = shallow(<ErrorList {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('Contains all items', () => {
    const wrapper = shallow(<ErrorList {...props} />);
    props.errors.map(errorMessage => expect(wrapper.text()).toContain(errorMessage));
  });
});
