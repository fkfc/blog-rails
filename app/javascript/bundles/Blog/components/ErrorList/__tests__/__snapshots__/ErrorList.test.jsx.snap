// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Post Component Should render correctly 1`] = `
<div
  className="alert alert-danger"
>
  <h5>
    4 errors prevented this test from saving:
  </h5>
  <ul>
    <li
      key="error0"
    >
      error1
    </li>
    <li
      key="error1"
    >
      error2
    </li>
    <li
      key="error2"
    >
      error3
    </li>
    <li
      key="error3"
    >
      error4
    </li>
  </ul>
</div>
`;
